src/main.c

```diff
@@ -30,23 +61,45 @@ print_error(char *prefix, rustls_result result)
   fprintf(stderr, "%s: %.*s\n", prefix, (int)n, buf);
 }
 
+#ifdef _WIN32
+const char *ws_strerror (int err)
+{
+  static char ws_err[50];
+
+  if (err >= WSABASEERR) {
+    snprintf(ws_err, sizeof(ws_err), "Winsock err: %d", err);
+    return ws_err;
+  }
+  /* Assume a CRT error */
+  return (strerror)(err);
+}
+#endif
+
 /*
  * Write n bytes from buf to the provided fd, retrying short writes until
  * we finish or hit an error. Assumes fd is blocking and therefore doesn't
  * handle EAGAIN. Returns 0 for success or 1 for error.
+ *
+ * For Windsock we cannot use a socket-fd in write().
+ * Call send() if fd > STDOUT_FILENO.
```

Comment:
I think `send` and `recv` are supported on both Linux and macOS. I'd be happier splitting this function into two: `write_all` (for FDs) and `send_all` (for sockets).

Reply:
That's what I was thinking too. But that would be too many diffs.